docs/src/examples.md

```diff
@@ -0,0 +1,609 @@
+# and pass the opened IO object to `CSV.File`, which will read the entire
+# input into a temporary file, then parse the data from the temp file
+file = CSV.File(open("iso8859_encoded_file.csv", enc"ISO-8859-1"))
+
+# to instead have the encoding conversion happen in memory, pass
+# `buffer_in_memory=true`; this can be faster, but obviously results
+# in more memory being used rather than disk via a temp file
+file = CSV.File(open("iso8859_encoded_file.csv", enc"ISO-8859-1"); buffer_in_memory=true)
+```
+
+### [Gzipped input](@id gzipped_input)
+
+```julia
+# assume I have csv text data compressed via gzip
+# no additional packages are needed; CSV.jl can decompress automatically
+using CSV
+
+# pass name of gzipped input file directly; data will be decompressed to a
+# temporary file, then mmapped as a byte buffer for actual parsing
+file = CSV.File("data.gz")
+
+# to instead have the decompression happen in memory, pass
+# `buffer_in_memory=true`; this can be faster, but obviously results
+# in more memory being used rather than disk via a temp file
+file = CSV.File("data.gz"; buffer_in_memory=true)
+```
+
+### [Delimited data in a string](@id csv_string)
+
+```julia
+using CSV
+
+# I have csv data in a string I want to parse
+data = """
+a,b,c
+1,2,3
+4,5,6
+"""
+
+# Calling `IOBuffer` on a string returns an in-memory IO object
+# of the string data, which can be passed to `CSV.File` for parsing
+file = CSV.File(IOBuffer(data))
+```
+
+### [Data from the web/a url](@id http)
+
+```julia
+# assume there's delimited data I want to read from the web
+# one option is to use the HTTP.jl package
+using CSV, HTTP
+
+# I first make the web request to get the data via `HTTP.get` on the `url`
+http_response = HTTP.get(url)
+
+# I can then access the data of the response as a `Vector{UInt8}` and pass
+# it directly to `CSV.File` for parsing
+file = CSV.File(http_response.body)
+
+# another option, with Julia 1.6+, is using the Downloads stdlib
+using Downloads
+http_response = Downloads.download(url)
+
+# by default, `Downloads.download` writes the response data to a temporary file
+# which can then be passed to `CSV.File` for parsing
+file = CSV.File(http_response)
+```
+
+#### [Reading from a zip file](@id zip_example)
+
+```julia
+using ZipFile, CSV, DataFrames
+
+a = DataFrame(a = 1:3)
+CSV.write("a.csv", a)
+
+# zip the file; Windows users who do not have zip available on the PATH can manually zip the CSV
+# or write directly into the zip archive as shown below
+;zip a.zip a.csv
+
+# alternatively, write directly into the zip archive (without creating an unzipped csv file first)
+z = ZipFile.Writer("a2.zip")
+f = ZipFile.addfile(z, "a.csv", method=ZipFile.Deflate)
+a |> CSV.write(f)
+close(z)
+
+# read file from zip archive
+z = ZipFile.Reader("a.zip") # or "a2.zip"
+
+# identify the right file in zip
+a_file_in_zip = filter(x->x.name == "a.csv", z.files)[1]
+
+a_copy = CSV.File(read(a_file_in_zip)) |> DataFrame
```

Comment:
probably `z` needs to be closed again (just for cleanness) - right?

Reply:
added